geth/node/status_node.go

```diff
@@ -87,53 +86,45 @@ func (n *StatusNode) Start(config *params.NodeConfig, services ...node.ServiceCo
 		return ErrNodeRunning
 	}
 
-	if err := n.createNode(config); err != nil {
-		return err
-	}
-	n.config = config
+	var err error
 
-	if err := n.start(services); err != nil {
+	statusDB, err := db.Create(config.DataDir, params.StatusDatabase)
+	if err != nil {
 		return err
 	}
 
-	if err := n.setupRPCClient(); err != nil {
+	n.db = statusDB
+
+	defer func() {
```

Comment:
Can we maybe create the database as the last step? It's easy to write 
```go
if err := someCall(); err != nil {
  return err
}
```
in this long method which will fail to clean it up.

Reply:
I need the db for both peer pool and the node creation.
I think I'd rather wrap this into another method, like

```go
func (n *StatusNode) start(config) error {
       db, err := leveldb.Open(....)
       if err != nil {
                 return err
       }
  
       err = n.startWithDB(config, db)
       if err != nil {
           db.Close()
       }
       return err
}
```

wdyt?

Reply:
That looks good!